src/main/java/org/assertj/core/api/Char2DArrayAssert.java

```diff
@@ -0,0 +1,178 @@
+  @Override
+  public Char2DArrayAssert isNotEmpty() {
+    arrays.assertNotEmpty(info, actual);
+    return myself;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Char2DArrayAssert hasSize(int expectedFirstDimensionSize, int expectedSecondDimensionSize) {
+    arrays.assertHasSize(info, actual, expectedFirstDimensionSize, expectedSecondDimensionSize);
+    return myself;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Char2DArrayAssert hasSameSizeAs(Object other) {
+    arrays.assertHasSameSizeAs(info, actual, other);
+    return myself;
+  }
+
+  /**
+   * Verifies that the actual array contains the given char[] at the given index.
+   * <p>
+   * Example:
+   * <pre><code class='java'> // assertion will pass
+   * assertThat(new char[][] {{'a', 'b'}, {'c', 'd'} }).contains(new char[] {'a', 'b'}, atIndex(O));
+   *
+   * // assertion will fail
+   * assertThat(new char[][] {{'a', 'b'}, {'c', 'd'} }).contains(new char[] {'a', 'b'}, atIndex(1));</code></pre>
+   *
+   * @param value the value to look for.
+   * @param index the index where the value should be stored in the actual array.
+   * @return myself assertion object.
+   * @throws AssertionError if the actual array is {@code null} or empty.
+   * @throws NullPointerException if the given {@code Index} is {@code null}.
+   * @throws IndexOutOfBoundsException if the value of the given {@code Index} is equal to or greater than the size of
+   *           the actual array.
+   * @throws AssertionError if the actual array does not contain the given value at the given index.
+   */
+  public Char2DArrayAssert contains(char[] value, Index index) {
+    arrays.assertContains(info, actual, value, index);
+    return myself;
+  }
+
+  /**
+   * Verifies that the actual array does not contain the given char[] at the given index.
+   * <p>
+   * Example:
+   * <pre><code class='java'> // assertion will pass
+   * assertThat(new char[][] {{'a', 'b'}, {'c', 'd'} }).doesNotContain(new char[] {'a', 'b'}, atIndex(1));
+   *
+   * // assertion will fail
+   * assertThat(new char[][] {{'a', 'b'}, {'c', 'd'} }).doesNotContain(new char[] {'a', 'b'}, atIndex(O));</code></pre>
+   *
+   * @param value the value to look for.
+   * @param index the index where the value should be stored in the actual array.
+   * @return myself assertion object.
+   * @throws AssertionError if the actual array is {@code null}.
+   * @throws NullPointerException if the given {@code Index} is {@code null}.
+   * @throws AssertionError if the actual array contains the given value at the given index.
+   */
+  public Char2DArrayAssert doesNotContain(char[] value, Index index) {
+    arrays.assertDoesNotContain(info, actual, value, index);
+    return myself;
+  }
+
+  /**
+   * Use unicode character representation instead of standard representation in error messages.
+   * <p>
+   * With standard error message:
+   * <pre><code class='java'> assertThat("a6c".toCharArray()).isEqualTo("abó".toCharArray());
```

Comment:
examples don't show char[][] which is a bit confusing

Reply:
done